app/client/src/features/events/Questions/AskQuestion/AskQuestion.tsx

```diff
@@ -37,20 +38,36 @@ export const ASK_QUESTION_MUTATION = graphql`
     }
 `;
 
+function isURL(str: string) {
+    const pattern = new RegExp('^(https?:\\/\\/)?'+
+      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ 
+      '((\\d{1,3}\\.){3}\\d{1,3}))'+ 
+      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ 
+      '(\\?[;&a-z\\d%_.~+=-]*)?'+ 
+      '(\\#[-a-z\\d_]*)?$','i'); 
+    return !!pattern.test(str);
+}
+
 function AskQuestion({ className, eventId }: AskQuestionProps) {
     const [isOpen, open, close] = useResponsiveDialog();
     const [user] = useUser();
     const [commit] = useMutation<AskQuestionMutation>(ASK_QUESTION_MUTATION);
     const { displaySnack } = useSnack();
 
     function handleSubmit(form: TQuestionFormState) {
-        commit({
-            variables: { input: { ...form, eventId, isFollowUp: false, isQuote: false } },
-            onCompleted(payload) {
-                if (payload.createQuestion.isError) displaySnack('Something went wrong!');
-                else close();
-            },
-        });
+        try {
+            if (form.question.length >= 1000) throw new ValidationError('Question is too long!')
```

Comment:
Hmm, this is an important business rule (no questions greater than 1k characters).  I almost think this should be an env variable.  

For example, think about a scenario where someone asks us to update the max question length.  We'd have to remember it's in this file as opposed to just using an env variable.  This comment is more so a note to myself, no changes on your part are required, but feel free to have input on what you think is best.

Reply:
I totally agree; In that case, I would replace the 1000 with something like MAX_QUESTION_LENGTH and declare that env variable in the .env file; I would also need to declare that env variable in docker if I'm not mistaken, correct?

Reply:
Yeah, that would be correct.  However, typically the .env is used for things like the URL to connect to the database or an API key.  I think the .env file could get cluttered if we start putting all of our business rules in there.

Maybe the solution would be to make a file like `rules.ts` where we just export a bunch of hard-coded variables.  So for example `export const QUESTIONS_MAX_LENGTH = 1000`. Then, we can import it like so `import { QUESTIONS_MAX_LENGTH } from '@local/rules';`